src/wfs/src/test/java/org/geoserver/wfs/TransactionTest.java

```diff
@@ -1095,4 +1095,34 @@ public void testBrokenDelete() throws Exception {
             fail("Should not throw an exception");
         }
     }
+
+    @Test
+    public void testBrokenUpdate() {
+
+        // perform an update
+        String update =
+                "<wfs:Transaction service=\"WFS\" version=\"1.0.0\" "
+                        + "xmlns:cgf=\"cite\" "
+                        + "xmlns:ogc=\"http://www.opengis.net/ogc\" "
+                        + "xmlns:wfs=\"http://www.opengis.net/wfs\" "
+                        + "xmlns:gml=\"http://www.opengis.net/gml\"> "
+                        + "<wfs:Update typeName=\"cgf:Polygons\" > "
+                        + "<wfs:Property>"
+                        + "<wfs:Name>id</wfs:Name>"
+                        + "<wfs:Value>t0003</wfs:Value>"
+                        + "</wfs:Property>"
+                        + "<ogc:Filter>"
+                        + "<ogc:PropertyIsEqualTo>"
+                        + "<ogc:PropertyName>id</ogc:PropertyName>"
+                        + "<ogc:Literal>t0002</ogc:Literal>"
+                        + "</ogc:PropertyIsEqualTo>"
+                        + "</ogc:Filter>"
+                        + "</wfs:Update>"
+                        + "</wfs:Transaction>";
+        try {
+            postAsDOM("wfs", update);
```

Comment:
You can check the contents of a service exception using one of the various ``checkOwsXYException`` methods.. I think you need ``checkOws10Exception`` but not completely sure.

Reply:
So you can :-p

Reply:
@ianturton this does not look like a serious test. If you're doing this on volunteer time, please wait and I'll add a better one (hopefully this weekend). If you are on consultancy time, please remember to charge for writing proper tests too.

Reply:
All I want to test is that it doesn't throw a NPE and we get a proper error message - seems like a fine test to me.

Reply:
You're sending a request to the service... even in case of a NPE, you should be getting a 200, with a service exception, telling inside the XML that a NPE happened. Or did the NPE occur inside the Dispatcher? (that's the only place where I think an exception would be able to escape out to the container).